app/src/main/java/com/klinker/android/twitter/activities/compose/Compose.java

```diff
@@ -1040,95 +1050,152 @@ public void run() {
+                    if (waitForLocation()) {
+                        Location location = mLastLocation;
+                        GeoLocation geolocation = new GeoLocation(location.getLatitude(), location.getLongitude());
+                        media.setLocation(geolocation);
+                    }
+                }
+                twitter.updateStatus(media);
+            }
+        }
+
         protected Boolean doInBackground(String... args) {
             status = args[0];
             try {
                 Twitter twitter = Utils.getTwitter(getApplicationContext(), settings);
                 Twitter twitter2 = Utils.getSecondTwitter(getApplicationContext());
 
-                if (remaining < 0 && !pwiccer) {
+                if (remaining < 0 && !pwiccer && !multiTweets) {
                     // twitlonger goes here
-
                     boolean isDone = false;
-
                     if (useAccOne) {
-                        TwitLongerHelper helper = new TwitLongerHelper(text, twitter);
-
-                        if (notiId != 0) {
-                            helper.setInReplyToStatusId(notiId);
-                        }
-
-                        if (addLocation) {
-                            if (waitForLocation()) {
-                                Location location = mLastLocation;
-                                GeoLocation geolocation = new GeoLocation(location.getLatitude(), location.getLongitude());
-                                helper.setLocation(geolocation);
-                            }
-                        }
-
-                        if (helper.createPost() != 0) {
-                            isDone = true;
-                        }
+                        isDone = tweetUsingTweetLonger(twitter);
                     }
 
                     if (useAccTwo) {
-                        TwitLongerHelper helper = new TwitLongerHelper(text, twitter2);
-
-                        if (notiId != 0) {
-                            helper.setInReplyToStatusId(notiId);
-                        }
-
-                        if (addLocation) {
-                            waitForLocation();
+                        isDone = tweetUsingTweetLonger(twitter2);
+                    }
 
-                            if (waitForLocation()) {
-                                Location location = mLastLocation;
-                                GeoLocation geolocation = new GeoLocation(location.getLatitude(), location.getLongitude());
-                                helper.setLocation(geolocation);
-                            }
+                    return isDone;
+                } else if (multiTweets && remaining < 0) {
+                    /* We need to schedule this activity because we are hitting the
+                    *  rate limit quite easily. We will hardcode to schedule to tweeet every
+                    *  minute */
+                    List<String> multiTweets = getMultipeTweets(status);
```

Comment:
This isn't going to work quite the way you think it is going to, with Twitter's new reply style.

Say that I am replying to you:

```
@wolverine2k really long text here, more than 140 characters!
```

If it is a reply, twitter will no longer count the @mentions in the tweet count. The issue here is that you are including those mentions in the tweet count, so when they are broken apart, if we had a long tweet, the first part would look like:

```
@mentions long multipart text here (1/2)
finishing the multipart tweets (2/2)
```

For a reply tweet (only a reply! If this is a brand new tweet, then the above format would be great!), I would have expected it to look like this:

```
@mentions part 1 (1/2)
@mentions part 2 (2/2)
```

So, if it is a reply, you will need to pull out those @mentions from the tweet text and pass the text, without them, into this `getMultipleTweets` method. Performance isn't a huge concern here, you could use a simple for loop to pull out the @mentions at the start of the tweet:

```java
if (notiId == 0) {
    // send the whole text, like you are currently doing
    getMultipleTweets(status);
} else {
    String mentions = "";
    for (String part : status.split(" ")) {
        if (part.contains("@")) {
            // we want to store all the mentions in the front of a reply
            mentions += part + " "; 
        } else {
            // exit the loop, we only want tweets that start with @mention. If it is later in the tweet, it still counts toward the 140 characters.
            break;
        }
    }

    getMultipleTweets(status.replace(mentions, ""));
}
```

You will have to test that code, I don't know if it compiles as is. But the logic should be clear. After you have done this, on any part of this reply, you will need to add the `reply.setAutoPopulateReplyMetadata(true)`, in the function that sends the plain text tweet. You will need to change the logic there, as well, because right now it will only set that flag if you have @mentions in the tweet text that is being sent to that function. By breaking it apart with the code I have provided here, you would get the tweet text without the mention in it:

```
@lukeklinker tweet part 1 here tweet part 2 here 

would get converted into:

tweet part 1 here (1/2)
tweet part 2 here (2/2)
```

Hopefully that makes sense. It is good stuff, you just overlooked this detail!